docs/docs/30-how-to-guides/16-working-with-warehouses.md

```diff
@@ -0,0 +1,161 @@
+---
+description: Learn how to work effectively with Warehouses
+sidebar_label: Working with Warehouses
+---
+
+# Working with Warehouses
+
+Each Kargo warehouse is represented by a Kubernetes resource of type
+`Warehouse`.
+
+## The `Warehouse` Resource Type
+
+A `Warehouse` resource's most important field is its `spec.subscriptions` field,
+which is used to subscribe to one or more:
+
+* Container image repositories
+
+* Git repositories
+
+* Helm charts repositories
+
+The following example shows a `Warehouse` resource that subscribes to a
+container image repository and a Git repository:
+
+```yaml
+apiVersion: kargo.akuity.io/v1alpha1
+kind: Warehouse
+metadata:
+    name: my-warehouse
+    namespace: kargo-demo
+spec:
+    subscriptions:
+    - image:
+        repoURL: public.ecr.aws/nginx/nginx
+        semverConstraint: ^1.26.0
+    - git:
+        repoURL: https://github.com/example/kargo-demo.git
+```
+:::info
+Kargo uses [semver](https://github.com/masterminds/semver#checking-version-constraints) to handle semantic versioning constraints.
+:::
+
+### Image Subscription
+
+For subscriptions to container image repositories, the `imageSelectionStrategy` field specifies the method for selecting
+the desired image. The available strategies for subscribing to an image repository are:
+
+- `Digest`: Selects the image with the specified digest.
+- `Lexical`: Selects the image with the lexicographically greatest tag.
+- `NewestBuild`: Selects the image with the most recent build time.
+- `SemVer`: Selects the image that best matches a semantic versioning constraint.
+
+### Git Subscription
+
+In subscriptions to Git repositories, the `commitSelectionStrategy` field
+specifies the method for selecting the desired commit.
+The available strategies for subscribing to a git repository are:
+
+- `Lexical`: Selects the commit with the lexicographically greatest reference.
+- `NewestFromBranch`: Selects the most recent commit from a specified branch.
+- `NewestTag`: Selects the most recent commit associated with a tag.
```

Comment:
Not quite what it does.

Unlike branches, tags are supposed to be immutable, so there should only be _one_ commit associated with each tag.

The purpose of this strategy is to select the commit associated with the newest (most recently created) tag.

Again, constraining the tags that are eligible is a good idea.